src/org/zaproxy/zap/extension/ascan/ActiveScan.java

```diff
@@ -289,7 +296,27 @@ public void run() {
             });
         }
 	}
-	
+
+    private void removeHistoryReference(HistoryReference remRef) {
+        if (View.isInitialised()) {

```

Comment:
Worth refactoring these methods. This check could be moved to the above if `hRef != null`, no longer needing this method and `addHistoryReference`, then call directly the "inEdt" methods. Also, the check `EventQueue.isDispatchThread()` in the "inEdt" methods can be removed (the thread notifying the new message is not the EDT).